packages/scoped-custom-element-registry/src/scoped-custom-element-registry.ts

```diff
@@ -391,7 +475,7 @@ if (!ShadowRoot.prototype.createElement) {
       // Approximate observedAttributes from the user class, since the stand-in element had none
       definition.observedAttributes.forEach((attr) => {
         if (instance.hasAttribute(attr)) {
-          definition.attributeChangedCallback.call(
+          definition.attributeChangedCallback!.call(
```

Comment:
I think the non-null assertion is ok in this case - my main question is why TS thinks this may be null. I want to do another pass to use optional-chaining and rely on null-checks a bit more. I'll come back to it.

Reply:
My guess is that it can't prove that `definition.attributeChangedCallback` hasn't changed between the first and second property access, since `forEach` and `hasAttribute` might be overridden to do something weird like change the value themselves or delay running the callback until some arbitrary later time. IDK but I doubt TS has special knowledge of / takes into consideration the implementation of built in functions during type checking.

Reply:
It's because we're using forEach instead of `for of`, TypeScript doesn't know when this callback will be called, and whether `definition` will be mutated in the meanwhile